extension/src/experiments/columns/model.ts

```diff
@@ -108,6 +110,40 @@ export class ColumnsModel extends PathSelectionModel<Column> {
     )
   }
 
+  private findChildrenColumns(
+    parent: string,
+    columns: Column[],
+    childrenColumns: string[]
+  ) {
+    const filteredColumns = columns.filter(
+      ({ parentPath }) => parentPath === parent
+    )
+    for (const column of filteredColumns) {
+      if (column.hasChildren) {
+        this.findChildrenColumns(column.path, columns, childrenColumns)
+      } else {
+        childrenColumns.push(column.path)
+      }
+    }
+  }
+
+  private getColumnsFromType(type: string): string[] {
+    const childrenColumns: string[] = []
+    const dataWithType = this.data.filter(({ path }) => path.startsWith(type))
+    this.findChildrenColumns(type, dataWithType, childrenColumns)
+    return childrenColumns
+  }
+
+  private getColumnOrderFromData() {
```

Comment:
Originally, I just went through the columns data and collected all columns that don't have children, but that causes some discrepancies.

To make sure that the columns collected from the data were in the correct order, I ended up needing to write a couple of different functions. But maybe there is a less complex way 🤔

Reply:
So this is the original `columnOrderState` right? Why not set it to the once so that it doesn't need to get recalculated every time?

Reply:
Done!